packages/ketcher-core/src/domain/entities/sgroup.ts

```diff
@@ -218,21 +227,94 @@ export class SGroup {
     this.pp = topLeftPoint
   }
 
-  getAttAtomId(struct: Struct): number {
-    for (const atomId of this.atoms) {
-      const atom = struct.atoms.get(atomId)
-      if (!atom) continue
-      if (Number.isInteger(atom.attpnt)) return atomId
+  isGroupAttached(struct: Struct): boolean {
+    return this.attachmentPoints.some((attachPoint) =>
+      this.isAttachmentPointAttached(struct, attachPoint)
+    )
+  }
+
+  addAttachmentPoint(attachmentPoint: SGroupAttachmentPoint): void {
+    const isAttachmentPointAlreadyExist = this.attachmentPoints.some(
+      ({ atomId }) => attachmentPoint.atomId === atomId
+    )
+
+    if (isAttachmentPointAlreadyExist) {
+      throw new Error(
+        'The same attachment point cannot be added to an S-group more than once'
+      )
     }
-    // in normal circumstances this should never be invoked
-    return this.atoms[0]
+
+    this.attachmentPoints.push(attachmentPoint)
   }
 
-  isGroupAttached(struct: Struct): boolean {
-    const attachPointId = this.getAttAtomId(struct)
-    const neighbours = struct.atomGetNeighbors(attachPointId)
+  addAttachmentPoints(
+    attachmentPoints:
+      | ReadonlyArray<SGroupAttachmentPoint>
+      | SGroupAttachmentPoint[]
+  ): void {
+    for (const attachmentPoint of attachmentPoints) {
+      this.addAttachmentPoint(attachmentPoint)
+    }
+  }
+
+  removeAttachmentPoint(attachmentPointAtomId: number): boolean {
+    const index = this.attachmentPoints.findIndex(
+      ({ atomId }) => attachmentPointAtomId === atomId
+    )
+    if (index !== -1) {
+      this.attachmentPoints.splice(index, 1)
+      return true
+    }
+    return false
+  }
+
+  getAttachmentPoints(): ReadonlyArray<SGroupAttachmentPoint> {
+    return this.attachmentPoints
+  }
 
-    return !neighbours?.every(({ aid }) => this.atoms.includes(aid))
+  getAttachmentPointsCount(): number {
+    return this.attachmentPoints.length
+  }
+
+  /**
+   * Why only one?
+   * Currently other parts of application doesn't support several attachment points for sgroup.
+   * So to support it - it's require to refactor almost every peace of code with sgroups.
+   *
+   *
+   * Why return 'undefined' without rollback?
+   * If sgroup doesn't have attachment points it can't be attached, (salt and solvents for example).
+   */
```

Comment:
```suggestion
  /**
   * Why only one?
   * Currently other parts of application don't support several attachment points for sgroup.
   * So to support it - it's required to refactor almost every peace of code with sgroups.
   *
   *
   * Why return 'undefined' without fallback?
   * If sgroup doesn't have attachment points it can't be attached, (salt and solvents for example).
   */
```